server/routers/index.js

```diff
@@ -1,8 +1,113 @@
 const express = require('express')
 const router = new express.Router()
 
-router.get('', (req, res) => {
+const middleware = require('../middleware/auth') 
+const User = require('../models/user') // User Model
+
+const { google } = require('googleapis')  /* Library for generating Auth Url and  generating Access Token from Auth Code  */
+const { OAuth2Client } = require('google-auth-library') /* Library Used for authenticating Access Token generated by Library Above */
+
+/* Accessing Credential provided by Google to initial Google Login */
+const CLIENT_ID = process.env.CLIENT_ID
+const CLIENT_SECRET = process.env.CLIENT_SECRET
+const REDIRECT_URL = process.env.REDIRECT_URI
+
+/* New oAuth2Client using credential provided above */
+const oAuth2Client = new google.auth.OAuth2(CLIENT_ID, CLIENT_SECRET, REDIRECT_URL)
+
+const client = new OAuth2Client(CLIENT_ID)
+
+
+router.get('', (req, res) => {  
   res.render('index')
 })
 
+
+
+// Route for initialising login request
+router.get('/login', (req, res) => {
+  if(!req.cookies.jwtToken){

```

Comment:
If you want to check if user is already logged in then verify that the jwt token is valid along with checking that it exists.